iothub/service/src/Amqp/AmqpConnectionHandler.cs

```diff
@@ -95,26 +95,42 @@ internal async Task OpenAsync(CancellationToken cancellationToken)
 
                 AmqpSettings amqpSettings = CreateAmqpSettings();
 
+                TransportSettings transportSettings = null; //either WebSocket or TCP
                 if (_useWebSocketOnly)
                 {
-                    // Try only AMQP transport over WebSocket
-                    _transport = await CreateClientWebSocketTransportAsync(cancellationToken).ConfigureAwait(false);
+                    transportSettings = new WebSocketTransportSettings()
+                    {
+                        Uri = new Uri(_credential.HostName),
+                        Proxy = _options.Proxy,
+                    };
                 }
                 else
                 {
-                    TlsTransportSettings tlsTransportSettings = CreateTlsTransportSettings();
-                    var amqpTransportInitiator = new AmqpTransportInitiator(amqpSettings, tlsTransportSettings);
-                    try
-                    {
-                        _transport = await amqpTransportInitiator.ConnectAsync(cancellationToken).ConfigureAwait(false);
-                    }
-                    catch (Exception e) when (!(e is AuthenticationException))
+                    transportSettings = new TcpTransportSettings
                     {
-                        if (Logging.IsEnabled)
-                            Logging.Error(this, e, nameof(OpenAsync));
+                        Host = _credential.HostName,
+                        Port = _credential.AmqpEndpoint.Port,
+                    };
+                }
 
-                        throw;
-                    }
+                var tlsTransportSettings = new TlsTransportSettings(transportSettings)
+                {
+                    TargetHost = _credential.HostName,
```

Comment:
do we need to set this in tcp settings again if we're setting it up here anyway?

Reply:
We do need to set it in both places otherwise the AMQP library throws